src/components/apparatusLists/ActionSequenceBox.tsx

```diff
@@ -41,26 +42,71 @@ const ListBoxScroller = styled.div`
 `;
 
 const ActionSequenceBox = ({ actionList, removeAction }) => {
+
+  //Setting the array from the parameter with hook.
+  const [list,setList] = useState(actionList);
+  if (list.length != actionList.length) {
+    setList(actionList);
+  }
+
+  //Settig the postion of the item dragged and dropped.
+  function handleOnDragEnd (result) {
```

Comment:
it's better to perform changes to the action list from overlay which is the parent component. for example, to remove an action from list, a function in overlay is called. doing this also removes the need to define your state variables and using props for setting the variables

Reply:
Thank you for your review! I made changes based on your suggestion. Can you please test it out and let me know if your still having the issue?

Reply:
Tested. It works

Reply:
Thanks!